extensions/resteasy/deployment/src/main/java/io/quarkus/resteasy/deployment/ResteasyScanningProcessor.java

```diff
@@ -340,6 +344,13 @@ public void build(
             servletContextParams.produce(new ServletInitParamBuildItem("resteasy.servlet.mapping.prefix", path));
             if (appClass != null) {
                 servletContextParams.produce(new ServletInitParamBuildItem(JAX_RS_APPLICATION_PARAMETER_NAME, appClass));
+                final String finalAppClass = appClass; // horrible java :(
```

Comment:
Why not make the `appClass` variable final above? Looking at the code above, it would make it more readable.

That being said, I think @mkouba had another fix in mind.

Still not convinced it's a good idea to do things like that for **every** component.

Reply:
Are you sure things would be cleaner? 
I don't think that making `appClass` `final` will be cleaner, since it uses "coupled" to `app` (along with `path`). I mean we could obviously extract a method and create a container class to make things final, but the rest of the code base doesn't use that approach. WDYT?

Reply:
Isn't it just about changing things to:
```
        final String path;
        final String appClass;
        if (!app.isEmpty()) {
            AnnotationInstance appPath = app.iterator().next();
            path = appPath.value().asString();
            appClass = appPath.target().asClass().name().toString();
        } else {
            path = resteasyConfig.path;
            appClass = null;
        }
```

Just saying looking at the code in GH so I might be missing obvious.

Reply:
OMG... I am SO not used to having `final` variables without immediate initialization... You are of course correct! I guess this is something that every one used to but me :)